airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/java/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.java

```diff
@@ -194,6 +197,19 @@ protected Map<String, String> getFinalMetadataColumnNames() {
   public void setup() throws Exception {
     generator = getSqlGenerator();
     destinationHandler = getDestinationHandler();
+
+    // In a real sync, we only execute the setup code once.
+    // We're doing this here instead of in a BeforeAll method for two reasons:
+    // 1. destinationhandler/sqlgenerator are instance variables
+    // 2. Implementing classes might have their own BeforeAll methods, and junit doesn't guarantee
+    // execution order between BeforeAll methods.
+    synchronized (SETUP_LOCK) {
+      if (!setupSqlCompleted) {
+        destinationHandler.execute(generator.setup());
+        setupSqlCompleted = true;
+      }
+    }
```

Comment:
Why not make the Generator's setup idempotent using an `AtomicBoolean isSetupDone`.  The UDF already ensures idempotency, rather than documenting setup should only be called once, may be make it only execute once no matter how many times it gets called.

Reply:
I should remove this stuff and just let it run multiple times >.> originally I was hoping to allow `setup` to be not threadsafe, then realized that we need to support concurrent syncs to the same destination

Reply:
which is why atomic boolean (or other in-connector tracking) doesn't solve the problem fully - if the user has two syncs scheduled to run at 10AM, both destination-postgres instances are going to try and create this function :(